cmd/mdmctl/config.go

```diff
@@ -80,21 +81,35 @@ func setCmd(cfg *ClientConfig, args []string) error {
 	return SaveClientConfig(cfg)
 }
 
-func clientConfigPath() (string, error) {
+func clientConfigDirPath() (string, error) {
 	usr, err := user.Current()
 	if err != nil {
 		return "", err
 	}
-	path := usr.HomeDir + "/.micromdm/default.json"
-	return path, nil
+	return path.Join(usr.HomeDir, ".micromdm"), nil
+}
+
+func clientConfigPath() (string, error) {
+	dir, err := clientConfigDirPath()
+	return path.Join(dir, "default.json"), err
 }
 
 func SaveClientConfig(cfg *ClientConfig) error {
-	path, err := clientConfigPath()
+	path, err := clientConfigDirPath()
+	if err != nil {
+		return err
+	}
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		err = os.MkdirAll(path, 0777)
+		if err != nil {
+			return err
+		}
+	}
+	path, _ = clientConfigPath()
```

Comment:
On second thought — even though that's true (there can never be an error here — it's probably best to check anyway, in case those functions get modified.